### DISCUSSION TITLE
Warning on ImGui enumerations

### DISCUSSION THREAD

**Discussion by guplasse** [ocornut/imgui#4147] [Old/Archived discussions]
Hello,
I'm using VS2019 and i have some warning on enumeration.

Warning code : C26812
Description : Type 'XXXX' is unbound. Prefer 'enum class' instead of 'enum'.

Is there a way to remove it ? Or should i modify ImGui's code by myself ?

**Comment by ocornut** [comment]
This and half a dozen of VS2019 static analyzer warnings and false positives have been fixed 17 days ago, see https://github.com/ocornut/imgui/pull/4073#issuecomment-829523422

**Reply by guplasse** [reply]
I take the lastest update (20/05/2021) and i have the same warning ...

**Reply by ocornut** [reply]
Could you post the exact lines and log? thank you

**Reply by guplasse** [reply]
Code

void UI::cUserInterface::ShowWindow_RenderPanel()
{
	ImGui::SetNextWindowSizeConstraints(ImVec2(0, -1), ImVec2(FLT_MAX, -1));      // Horizontal only

	float baseheight = (float)m_Engine->Height() - m_MenuHeight - ScenePanelHeight;
	float basewidth = leftpanelWidth;
	float x = (float)m_Engine->Width() - basewidth;
	ImGui::SetNextWindowPos(ImVec2(x, m_MenuHeight + ScenePanelHeight));
	ImGui::SetNextWindowSize(ImVec2(basewidth, baseheight));

	ImGuiWindowFlags flags = ImGuiWindowFlags_NoMove
		| ImGuiWindowFlags_NoCollapse;

	ImGui::PushStyleVar(ImGuiStyleVar_WindowBorderSize, 1.0f);
	ImVec2 Padding = ImGui::GetStyle().WindowPadding;
	Padding.y = 0.0f;
	ImGui::PushStyleVar(ImGuiStyleVar_WindowPadding, Padding);
	ImGui::PushStyleVar(ImGuiStyleVar_TabRounding, 0);
	ImGui::PushStyleVar(ImGuiStyleVar_ItemInnerSpacing, ImVec2(0.0f, 4.0f));

	// Main body of the Demo window starts here.
	ImGui::Begin("Rendu", NULL, flags);
	leftpanelWidth = ImGui::GetWindowSize().x;

	ImGui::PushStyleColor(ImGuiCol_Border, ImVec4(0.2f, 0.2f, 0.2f, 1.0f));
	ImGui::PushItemWidth(150.0f);
	ImGui::BeginTabBar("##tabs", ImGuiTabBarFlags_None);
	{
		if (ImGui::BeginTabItem("Moteur", NULL, flag_selected))
		{
			ImGui::PushStyleVar(ImGuiStyleVar_ItemInnerSpacing, ImVec2(4.0f, 4.0f));
			ShowTab_Engine();
			ImGui::EndTabItem();
			ImGui::PopStyleVar();
		}
		if (ImGui::BeginTabItem("Fond", NULL, flag_selected))
		{
			ImGui::PushStyleVar(ImGuiStyleVar_ItemInnerSpacing, ImVec2(4.0f, 4.0f));
			ShowTab_Background();
			ImGui::EndTabItem();
			ImGui::PopStyleVar();
		}

		if (select_mat)
		{
			flag_selected = ImGuiTabItemFlags_SetSelected; // this one return the warning
		}
		if (ImGui::BeginTabItem("Materiau", NULL, flag_selected))
		{
			ImGui::PushStyleVar(ImGuiStyleVar_ItemInnerSpacing, ImVec2(4.0f, 4.0f));
			ShowTab_Material();
			ImGui::EndTabItem();
			ImGui::PopStyleVar();
		}
		if (select_mat)
		{
			flag_selected = ImGuiTabItemFlags_None;
			select_mat = false;
		}
	}
	ImGui::EndTabBar();

	ImGui::PopStyleColor();
	ImGui::PopItemWidth();
	ImGui::End();

	ImGui::PopStyleVar(4);
}


Log :

Gravité	Code	Description	Projet	Fichier	Ligne	État de la suppression
Avertissement	C26812	Le type enum 'ImGuiTabItemFlags_’ est non délimité. Préférez ’enum class’ à ’enum' (enum.3).	3D_editor	D:\Depots\3D_editor\src\UI\UI.cpp	315

**Comment by ocornut** [comment]
Well that’s a warning in your code, not in dear imgui. You should not enable this warning in your code I suppose. It’s simply requesting to follow a certain coding style...

**Reply by guplasse** [reply]
OK, but i don't understant why it appears at this line 
--- flag_selected = ImGuiTabItemFlags_SetSelected;
but not at this one 
--- flag_selected = ImGuiTabItemFlags_None;
this two enumerators are from the same enum ...
